flang/lib/Optimizer/Builder/Runtime/Reduction.cpp

```diff
@@ -149,6 +149,22 @@ struct ForcedNorm2Real16 {
   }
 };
 
+/// Placeholder for real*16 version of Norm2Dim Intrinsic
+struct ForcedNorm2DimReal16 {
+  static constexpr const char *name = ExpandAndQuoteKey(RTNAME(Norm2DimReal16));
+  static constexpr fir::runtime::FuncTypeBuilderFunc getTypeModel() {
+    return [](mlir::MLIRContext *ctx) {
+      auto boxTy =
+          fir::runtime::getModel<const Fortran::runtime::Descriptor &>()(ctx);
+      auto strTy = fir::ReferenceType::get(mlir::IntegerType::get(ctx, 8));
+      auto intTy = mlir::IntegerType::get(ctx, 8 * sizeof(int));
```

Comment:
I would advise using `fir::runtime::getModel<const char*>` and `fir::runtime::getModel<int>` for `strTy` and `intTy` too to make any future work enabling cross compilation easier/centralize the work.

If this is the way it is done for other Forced... helper, you can keep it that way and we should update all usages in one go.

Reply:
Good idea! I created enhancement request https://github.com/llvm/llvm-project/issues/83290 for this.  We use direct types creation all over the place.